libs/MobileSync/MobileSync/Classes/Target/SFSyncUpTarget.m

```diff
@@ -276,10 +276,10 @@ - (void)sendRequest:(SFRestRequest*)request
        completionBlock:(SFSyncUpTargetCompleteBlock)completionBlock
              failBlock:(SFSyncUpTargetErrorBlock)failBlock
 {
-    [SFMobileSyncNetworkUtils sendRequestWithMobileSyncUserAgent:request failBlock:^(NSError *e, NSURLResponse *rawResponse) {
+    [SFMobileSyncNetworkUtils sendRequestWithMobileSyncUserAgent:request failureBlock:^(id response, NSError *e, NSURLResponse *rawResponse) {
         self.lastError = e.description;
```

Comment:
lastError should get the body of the response (if available) instead of e.description.
That would address issues like https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/issues/162

Reply:
Would that be considered a breaking change for this point release, even though it doesn't break public API? Should we do that in `9.0`?

Reply:
May be.
Although, the shape for lastError has never been well defined (the body of the response is in there but mixed up in the error message) and can changed depending of the NSError we build upstream.
Devs complained about the fact that is cannot be easily parsed. To extract any useful information, one has to do "string contains" type of logic - that logic might still work if we provide the clean body.

Reply:
That change could/should be done in a separate dedicated PR.

Reply:
True - @bbirman is going to look into this.